File: ass1_2_a.cpp
#include <bits/stdc++.h>
#include <iostream>
using namespace std;
int main()
{
    clock_t start, end;
  
    start = clock();
    int n;
    cin >> n;
    int M1[n][n];
    int M2[n][n];
    int M[n][n];
    for (int i = 0; i < n; i++){
        
        for (int j = 0; j < n; j++){
            
            M1[i][j] = rand() % 10;
            M2[i][j] = rand() % 10;
        }
    }
 
    for (int i = 0; i < n; i++){
        
        for (int j = 0; j < n; j++){
            
            int p = 0;
            for (int k = 0; k < n; k++){
               
                p += M1[i][k] * M2[k][j];
            }
            M[i][j] = p;
        }
    }
    end = clock();
    double time_taken = double(end - start) / double(CLOCKS_PER_SEC);
    cout << "Time taken by program is : " << fixed 
         << time_taken << setprecision(5);
    cout << " sec " << endl;
    return 0;
}

File: ass1_2_a.py
import time
import random
start = time.time()
n=int(input())
M1=[[0]*n]*n
M2=[[0]*n]*n
M=[[0]*n]*n
for i in range(n):
    for j in range(n):
        M1[i][j]=random.randint(0,10)
        M2[i][j]=random.randint(0,10)
for k in range(n):
    for l in range(n):
        p=0
        for m in range(n):
            p+=M1[k][m]*M2[m][l]
        M[k][l]=p
end = time.time()
print("The time of execution of above program is :", end - start)

















































